src/components/timecard/edit-shift-hours/EditShiftHours.jsx

```diff
@@ -111,19 +142,79 @@ const EditShiftHours = ({
     return false;
   };
 
+  const setRedirectAndMessages = (timecardDate, startDate, endDate) => {
+    if (timecardDate && startDate && endDate) {
+      const currentDay = formatJustDay(timecardDate);
+      const startDay = formatJustDay(startDate);
+      const endDay = formatJustDay(endDate);
+
+      let datesMoved = false;
+
+      if (currentDay !== startDay) {
+        datesMoved = true;
+        setShouldRedirect(true);
+        setRedirectTarget(`/timecard/${startDate}`);
+      } else if (currentDay !== endDay) {
+        datesMoved = true;
+        setShouldRedirect(true);
+        setRedirectTarget(`/timecard/${endDate}`);
+      }
+
+      if (datesMoved) {
+        const formattedStart = formatDateNoYear(startDate);
+        const formattedEnd = formatDateNoYear(endDate);
+        summaryMessages['update'] = {
+          message: `Timecard has been updated to start ${formattedStart} and end ${formattedEnd}`,
+        };
+      } else {
+        const formattedTimecard = formatDateNoYear(timecardDate);
+        summaryMessages['update'] = {
+          message: `Timecard has been updated, it now starts and ends on ${formattedTimecard}`,
+        };
+      }
+      setSummaryMessages(summaryMessages);
+      setIsAlertVisible(true);
+    }
+  };
+
   const onSubmit = async (formData) => {
     dayjs.extend(utc);
 
-    const actualStartDate = formatDate(timecardDate);
+    let localStartDate =
+      timeEntry.startTime !== '' ? timeEntry.startTime : timecardDate;
```

Comment:
Is `timeEntry.startTime !== ''` just checking if a time entry already exists? 

In `EditShiftTimecard`, we use `const timeEntryExists = !!timeEntry?.startTime && timeEntry.startTime !== ''`. Wondering if it might be useful to create a similar constant to use in this file to make it easier to read?

Reply:
Constant added for start and end date